src/java.base/share/classes/jdk/internal/constant/MethodTypeDescImpl.java

```diff
@@ -137,24 +137,25 @@ private static ClassDesc[] paramTypes(String descriptor, int start, int end) {
                 throw badMethodDescriptor(descriptor);
             }
             if (paramCount < 8) {
-                // use 0 to indicate that the current parameter length is greater than 256 and needs to be reparsed
                 lengths = (lengths << 8) | (len > 0xFF ? 0 : len);
             }
             paramCount++;
             cur += len;
         }
 
         var paramTypes = new ClassDesc[paramCount];
-        int paramIndex = 0;
-        int lengthsParamEnd = Math.min(paramCount, 8) - 1;
+        int paramIndex = 0,
+            lengthsEnd = Math.min(paramCount, 8) - 1;
         for (int cur = start; cur < end; ) {
-            int len = 0,
-                num = lengthsParamEnd - paramIndex;
+            int len;
+            int num = lengthsEnd - paramIndex;
             if (num >= 0) {
                 int shift = num << 3;
-                len = (int) ((lengths & (0xFFL << shift)) >>> shift);
-            }
-            if (len == 0) {
+                len = (int) ((lengths & (0xFFL << shift)) >> shift) & 0xFF;
```

Comment:
The `0xFFL << shift` step is unnecessary with the trailing `& 0xFF`:
```suggestion
                len = (int) (lengths >>> shift) & 0xFF;
```

Reply:
Under MacBook M1 Pro, the performance of running the scenario `(Ljava/lang/Object;Ljava/lang/String;)I` multiple times is very different. The above code is rollback verification, but it is found that the problem is not solved.